x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/selectors.ts

```diff
@@ -299,3 +305,29 @@ export const searchBarQuery: (state: Immutable<EndpointState>) => Query = create
     return decodedQuery;
   }
 );
+
+export const getCurrentIsolationState = (
+  state: Immutable<EndpointState>
+): EndpointState['isolateHost'] => {
+  return state.isolateHost;
+};
```

Comment:
Do we think the naming here will get confusing once we add in the actual state of the endpoint (being isolated/not)?

To be extremely precise (if it helps disambiguating a name), these states are reflective of:

- not the endpoint itself (being currently isolated or not)
- not the endpoint's configuration (e.g. endpoint knowing it should be isolated, but currently isn't)
- not the action itself (which may still yet fail, error, or expire)

but only the _request to create an action_. The action-creation-request success/fail/err

Reply:
I'll try to think of better naming. Got any suggestions? 😄

Reply:
I suggest adding "request" in there somewhere. So maybe `*IsolationState` -> `*IsolationRequestState`?

So then the status of the actions themselves can be `*IsolationActionState`. 

that just leaves Endpoint's `current` + `configured` state pairs, but they have entries in `HostInfo` already, so they may not need as many vars dedicated. But "Current" and "Configured" can likely be the descriptors for those, if necessary.